### PR TITLE
新的量化压缩算法和示例

### LINKED ISSUE #742: 在 SecretFlow 中实现新的通讯量化压缩算法
> 此 ISSUE 为 [隐语开源共建计划（SecretFlow Open Source Contribution Plan，简称 SF OSCP）第二期](https://github.com/orgs/secretflow/discussions/741)任务 ISSUE，欢迎社区开发者参与共建～
> 若有感兴趣想要认领的任务，但还未报名，辛苦先完成[报名](https://www.wjx.top/vm/QKio0dq.aspx# )进行哈～

## 任务介绍

- 任务名称：在Secretflow中实现新的通讯量化压缩算法
- 技术方向：联邦学习和人工智能
- 任务难度：热身🌟

## 详细要求

- 功能性：在 SecretFlow 中实现以下功能代码以及实现过程文档，功能包括：
  - 参考现有的模型量化算法，在 SecretFlow 联邦学习中，实现对 embedding 和 gradient 的压缩；
  - 基于 Compressor 实现压缩代码，并对模型的拆分逻辑进行封装，提交代码到 secretflow/utils/compressor.py中
  - 需要同时提交 tutorial，example 代码
- 收敛性/准确性：包含 simulator 跑出的实验数据并且证明收敛性/准确性，与明文结果对比，AUC 差值不应超过 0.05
- 代码规范：Python 代码需要使用 black+isort 进行格式化（流水线包含代码规范检查卡点）
- 提交说明：
  - 关联该 issue 并提交代码至 https://github.com/secretflow/secretflow/tree/main/examples/app/v_recommendation
  - 整理完整实现过程为文档，并提交到 https://github.com/secretflow/secretflow/tree/main/docs/tutorial

## 能力要求
- 熟悉拆分学习，熟悉量化模型压缩算法
- 熟悉 TensorFlow 或者 PyTorch

### PR DESCRIPTION
**Pull request by ues1df**
# Preamble
Close #742 

本想实现google论文中的INT8整型量化压缩算法，但是拉取了1.1.0版本后发现已经被其他人实现了......
转而写了一个（伪）FP8的压缩方法，以及在示例文档中进行了使用。
示例文档中还实现了一个基于k-means压缩的方法，用于介绍怎么自行写compressor，但是8bit下（256个centroid）速度会很慢。

### CODE REVIEW

**Comment by ues1df** [comment]
I have read the CLA Document and I hereby sign the CLA
